modules/rostests/apitests/version/VerQueryValue.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Tests for VerQueryValue[A/W]
+ * COPYRIGHT:   Copyright 2024 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include "precomp.h"
+
+typedef struct
+{
+    WORD wLength;
+    WORD wValueLength;
+    WORD wType;
+    WCHAR szKey[16]; // L"VS_VERSION_INFO"
+    WORD Padding1;
+    VS_FIXEDFILEINFO Value;
+
+    struct _StringFileInfo
+    {
+        WORD wLength;
+        WORD wValueLength;
+        WORD wType;
+        WCHAR szKey[15]; // L"StringFileInfo"
+
+        struct _StringTable
+        {
+            WORD wLength;
+            WORD wValueLength;
+            WORD wType;
+            WCHAR szKey[9]; // L"000004B"
+
+            struct _String_CompanyName
+            {
+                WORD wLength;
+                WORD wValueLength;
+                WORD wType;
+                WCHAR szKey[12]; // L"CompanyName"
+                WORD Padding1;
+                WCHAR Value[16]; // L"ReactOS Project"
+            } CompanyName;
+
+            struct _String_Comments
+            {
+                WORD wLength;
+                WORD wValueLength; // 0
+                WORD wType;
+                WCHAR szKey[9]; // L"Comments"
+            } Comments;
+
+            struct _String_Comments2
+            {
+                WORD wLength;
+                WORD wValueLength; // 0
+                WORD wType;
+                WCHAR szKey[10]; // L"Comments2"
+                WORD Padding;
+            } Comments2;
+
+            struct _String_FooBar
+            {
+                WORD wLength;
+                WORD wValueLength;
+                WORD wType;
+                WCHAR szKey[7]; // L"FooBar"
+                WCHAR Value[4]; // L"Bar"
+            } FooBar;
+
+        } StringTable;
+    } StringFileInfo;
+} TEST_VERSIONINFO;
+
+static const TEST_VERSIONINFO g_VersionInfo =
+{
+    .wLength = 298, // Header: 40, Value: 54, StringFileInfo: 204
+    .wValueLength = sizeof(VS_FIXEDFILEINFO),
+    .wType = 0,
+    .szKey = L"VS_VERSION_INFO",
+    .Padding1 = 0,
+    .Value =
+    {
+        .dwSignature = 0xFEEF04BD,
+        .dwStrucVersion = 0x00010000,
+        .dwFileVersionMS = 0x00050002,
+        .dwFileVersionLS = 0x0ECE0F77,
+        .dwProductVersionMS = 0x00040009,
+        .dwProductVersionLS = 0x00040004,
+        .dwFileFlagsMask = 0x00000002,
+        .dwFileFlags = 0x00000000,
+        .dwFileOS = 0x00000004,
```

Comment:
Technically this is `VOS__WINDOWS32`, but I'm winding whether you could use `VOS_NT_WINDOWS32` there (equals to 0x00040004)